src/dynamic/pointstamp.rs

```diff
@@ -26,10 +26,20 @@ pub struct PointStamp<T> {
     pub vector: Vec<T>,
 }
 
-impl<T> PointStamp<T> {
+impl<T: Timestamp> PointStamp<T> {
     /// Create a new sequence.
+    ///
+    /// This method will modify `vector` to ensure it does not end with `T::minimum()`.
     pub fn new(vector: Vec<T>) -> Self {
-        PointStamp { vector }
+        let mut result = PointStamp { vector };
+        result.enforce();
+        result
+    }
+    /// Enforces that `self` not end with `T::minimum()` by popping elements until it is true.
+    pub fn enforce(&mut self) {
```

Comment:
Does `enforce` need to be `pub`? It seems the only way to construct a `PointStamp` is through `new`, and all other places that would modify the vector go through this API? In this case, nobody should need to explicitly call `enforce` I believe -- this is not true if there is an API that modifies the pointstamp such that it doesn't have the canonical representation anymore.

Oh, I see: This is relevant as long as `vector` is pub. If that changes, we can probably make this method private.

Reply:
Resolved as "`enforce` deleted".